Add proposed project page metadata for terrasos

wgardiner opened the pull request:
This PR adds a LinkML schema and example data for the proposed additional Terrasos fields for ProjectPage.

blushi commented on schema/data/project-page-data.yaml:
```diff
@@ -0,0 +1,48 @@
+socialCulturalIndex:
+  numericValue: 85.0
+  unit: unit:UNITLESS
+environmentalAuthority:
```
I'm wondering whether we should include the new project partners in here as well as shown on: https://www.figma.com/design/brkxGV5qNOkZUp0YQl1cxO/Terrasos-Phase-1?node-id=500-31174&node-type=frame&m=dev
Historically, we've been storing this data both as off-chain relationship on `project` table (`developer_id`, `verifier_id`) and in the on-chain project metadata.
I guess we need to decide here if we want to define new db relationships (and tables in case it's 1 to many) for those new roles, or just store them in jsonld for simplicity

wgardiner replied:
I've added `projectOperator`, `projectMonitor`, and `projectOwner`, @clevinson and I chatted about the existing `projectVerifier` and `projectDeveloper` fields and decided to omit them, so we can handle them with the existing `project` table solution, does this sound good to you @blushi ?

blushi replied:
Well historically, we put those stakeholders both in the on chain project metadata because this is important project info and as offchain project relationships to be able to map it to existing accounts if any. I guess we should keep things consistent here.
In any case, in the code, we always look at both sources of data to display those existing stakeholders.
So I would suggest that we keep the `projectVerifier` and `projectDeveloper` fields in the on chain metadata, to keep things consistent, and we can also provide the on chain project dedicated fields if there's any account profile we should link to.
I'm fine not building new db columns/tables to store the new `projectOperator`, `projectMonitor` and `projectOwner` for now to save time and if we think they won't map to actual account, so ok to have those new fields only in the project metadata.

blushi replied:
Also can there be multiple `projectMonitor`s? cc/ @clevinson @erikalogie

erikalogie replied:
@blushi I've asked Malu from Terrasos and our science team about this

erikalogie replied:
Malu says that it is possible to have more than one monitor, though extremely uncommon

wgardiner replied:
I just added back the `projectDeveloper` and `projectVerifier` fields to the metadata. For now I left `projectMonitor` as a single field

blushi reviewed (commented):
Should we run linkml [gen-doc](https://linkml.io/linkml/generators/markdown.html)?

paul121 commented:
Yes, @wgardiner if you could do the steps described in the schema README: https://github.com/regen-network/regen-data-standards/blob/main/schema/README.md

@blushi like I told Will though, I think we should wait on merging this PR until we make progress on some other MVP things as we still figure out the MVP structure of this framework site. This is a good start but don't put too much effort here until things are ready

clevinson commented on schema/data/project-page-data.jsonld:
```diff
@@ -0,0 +1,242 @@
+    }
+  },
+  "projectDuration": "P1Y",
+  "projectDurationMinimum": "P6M",
+  "projectDurationMaximum": "P2Y",
```
I'm not sure that we need minimum and maximum defined at the project level.  I believe this is diff btw protocol v3 and v4, and if that's the only thing that would change min & max duration, then maybe we can infer this from the protocol, and explicitly set the protocol version somewhere in the project page metadata?

blushi replied:
projectDurationMaximum/Minimum will be stored at the credit class metadata level so let's just add creditClassVersion in the project metadata

S4mmyb replied:
In v4 of the protocol, @clevinson @blushi @paul121, the min duration is 20 years, the max is 50. Technically they still could re-register so the max can be extended. I'm wondering though, if it makes more sense to put the duration min and max at the protocol level and place the project term here as they should specify a term in the project plan...

blushi replied:
> projectDurationMaximum/Minimum will be stored at the credit class metadata level 

This meant to put duration min/max at the protocol level indeed (but we just preferred to use the technical terms from Regen Ledger to be clear, ie credit class), while the project duration/term would be in the project metadata.
Is this what you meant? @S4mmyb

clevinson commented on schema/data/project-page-data.jsonld:
```diff
@@ -0,0 +1,242 @@
+  ],
+  "ecologicalConnectivityIndex": {
+    "numericValue": 75.5,
+    "unit": "unit:UNITLESS"
```
```suggestion
```

I think we can actually drop this, since we won't be ever having a unit for `ecologicalConnectivityIndex`

clevinson commented on schema/data/project-page-data.jsonld:
```diff
@@ -0,0 +1,242 @@
+  },
+  "socialCulturalIndex": {
+    "numericValue": 85.0,
+    "unit": "unit:UNITLESS"
```
Same as with `ecologicalConnectivityIndex`
```suggestion
```

clevinson commented on schema/data/project-page-data.jsonld:
```diff
@@ -0,0 +1,242 @@
+  },
+  "projectMonitor": {
+    "name": "EAFIT",
+    "showOnProjectPage": true,
```
```suggestion
```

Since all of these are defined in the project page metadata, I don't know if we need this explicitly set. Maybe @blushi has thoughts?

clevinson reviewed (commented):
This looks great. A few questions on my side that we can maybe discuss on Monday during Terrasos sync call. After that, happy to get this merged and iterate as more pieces come up.

wgardiner commented:
I've just updated this PR to address feedback from comments and the discussion this morning. Significantly, I moved the content of `project-page.yaml` into `project.yaml`, with the idea that this data can be represented in either place, project page metadata acting as an override for project metadata.

paul121 commented on schema/src/project.yaml:
```diff
@@ -15,23 +15,42 @@ default_range: string
 
 classes:
   Project:
+    tree_root: true
```
Is it important to set `tree_root` here? Presumably this helped to create the `project-data.*` files?

Just worry because it says only one class can be set as a root. We [import](https://linkml.io/linkml/schemas/imports.html) classes from multiple schemas which might cause an issue.

From https://linkml.io/linkml/intro/tutorial02.html:

> In LinkML, the tree_root slot is used to designate a class as the root of a tree structure. Only one class in a schema can be set as root. If more than one class is marked as tree_root, a validation error will occur.

> Marking one class to serve as the root of the tree (as “container” of the other classes) is especially important when serializing and deserializing data. The class marked as tree_root will be the top-level object in the serialized data.

wgardiner replied:
removed

paul121 commented on schema/src/project.yaml:
```diff
@@ -70,11 +139,143 @@ slots:
-        meaning: rfs:Organization
\ No newline at end of file
+        meaning: rfs:Organization
+  MarketTypeTypes:
+    description: The type of marketplace
+    permissible_values:
+      rfs:ComplianceMarket:
+        description: a compliance market
+      rfs:VoluntaryMarket:
+        description: a voluntary market
```
I think the `rfs:` prefix should only be included in the `meaning` property of the enum's permissible value. This way `rfs` will be expanded to the full URI from the context.  The key for each value should just be the text identifier.  See how `ProjectRoleTypes` is configured.

wgardiner replied:
done

paul121 commented on schema/src/project.yaml:
```diff
@@ -70,11 +139,143 @@ slots:
+    description: The management areas associated with the project.
+    multivalued: true
+    inlined: true
+  activity:
+    # TODO: associate with regen data standards activities
+    range: ActivityType 
+    description: the activity.
+    multivalued: true
```
What `slot_uri` do we want for this?

wgardiner replied:
just added it as `rfs:activity` explicityly, I think by default it'll take your key and prefix it with your `default_prefix`

paul121 commented on schema/src/project.yaml:
```diff
@@ -70,11 +139,143 @@ slots:
+    range: string
+    slot_uri: rfs:subWatershed
+    description: The sub-watershed associated with the project.
+  ecosystemType:
+    slot_uri: rfs:ecosystemType
+    range: EnvironmentTypeTypes
+    description: The type of ecosystem associated with the project.
+    multivalued: true
```
Are we intentionally using "ecosystem" and "environment type" terms in the same context here?

The current description for the Environment Type taxonomy says it should not be used to describe ecosystems: https://framework.regen.network/taxonomy/environment-type

> The classification is designed to provide a mental image of the landscape where project activities occur. These should not be rigorously defined technical definitions that would be used to describe ecosystems or land cover/land use mapping classes.

wgardiner replied:
Ah, I thought that we were already using `ecosystemType` in metadata elsewhere (project? credit class? having trouble finding it now) and that Ned has advised us to change our language from `ecosystem` to `environment`. So my intention here was to stay consistent with existing data, while referencing the data standards work. Do you have perspective or is there someone else we need to rope in?

clevinson replied:
@S4mmyb do you have thoughts here?

Summary is that in the existing old project metadata we used to use `ecosystemType`, but with Ned's work I think there was a desire to switch to `environmentType`, is that correct? Should we be changing all metadata for new projects to use `environmentType` instead of `ecosystemType` now?

S4mmyb replied:
Yeah it should be `environmentType` as opposed to `ecosystemType`. In general, whatever Ned had in the PR he made we should adopt and overwrite. At some point, we'll have to go back and fix all of the other metadata sets, but we can do that in the future I think. Unless it breaks the way the front end consumes this kind of data. @blushi if we used `environmentType` instead, would that affect our ability to render tags on the pages? If so we probably want to push this change to a future PR where we change everything across the board...

clevinson replied:
We can also have a front end PR that looks for `environmentType` and if not present, falls back to `ecosystemType`, this way we can take our time migrating legacy projects and that won't block us getting these new projects migrated to the new vocabulary.

blushi replied:
> We can also have a front end PR that looks for `environmentType` and if not present, falls back to `ecosystemType`, this way we can take our time migrating legacy projects and that won't block us getting these new projects migrated to the new vocabulary.

yes we can do what @clevinson proposes

S4mmyb commented:
> This PR adds a LinkML schema and example data for the proposed additional Terrasos fields for ProjectPage.

With that in mind, do we want to name these files something which specifies that it's Terrasos? We could always use the credit class ID, `BT01`

clevinson commented:
Agreed, Let's merge this in now and have a follow-up PR to break it out with inheritance so we can have Terrasos specific fields in a separate file?

wgardiner commented:
Just updated `ecosystemType` to `environmentType`
